operator/v1/types_machineconfiguration.go

```diff
@@ -55,8 +53,76 @@ type MachineConfigurationSpec struct {
 	// has no effect on cluster upgrades which will still incur node disruption where required.
 	// +optional
 	NodeDisruptionPolicy NodeDisruptionPolicyConfig `json:"nodeDisruptionPolicy"`
+
+	// bootImageSkewEnforcement allows an admin to set the behavior of the boot image skew enforcement mechanism.
+	// +openshift:enable:FeatureGate=BootImageSkewEnforcement
+	// +optional
+	BootImageSkewEnforcement BootImageSkewEnforcement `json:"bootImageSkewEnforcement"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.mode) && (self.mode == 'Automatic' || self.mode =='Manual') ?  has(self.clusterBootImage) : !has(self.clusterBootImage)",message="clusterBootImage is required when mode is Automatic or Manual, and forbidden otherwise"
+// +union
+type BootImageSkewEnforcement struct {
+	// mode determines the underlying behavior of skew enforcement mechanism.
+	// Valid values are Automatic, Manual and None.
+	// Automatic means that the MCO will store the OCP version associated with the last boot image update in the
+	// clusterBootImage field.
+	// Manual means that the cluster admin is expected to perform manual boot image updates and store OCP version
+	// associated with the last boot image update in the clusterBootImage field.
+	// In Automatic and Manual mode, the MCO will prevent upgrades when the boot image skew exceeds the
+	// skew limit described by the release image.
+	// None means that the MCO will permit upgrades when the boot image exceeds the skew limit
+	// described by the release image. This will affect the cluster's ability to scale.
+	// This field is required.
+	// +unionDiscriminator
+	// +required
+	Mode BootImageSkewEnforcementMode `json:"mode"`
+
+	// clusterBootImage describes the current boot image of the cluster. This will be used to enforce the skew limit.
+	// This value will be compared against the cluster's skew limit to determine skew compliance.
+	// Required when mode is set to "Automatic" or "Manual" and forbidden otherwise.
+	// +optional
+	ClusterBootImage *ClusterBootImage `json:"clusterBootImage,omitempty"`
 }
 
+// ClusterBootImage describes the boot image of a cluster. It stores the RHCOS version of the boot image and
+// the OCP release version which shipped with that RHCOS boot image.
+// If ocpVersion and rhcosVersion are defined, both values will be used for checking skew compliance.
+// If only ocpVersion is defined, only that value will be used for checking skew compliance.
```

Comment:
Could we also include some of this information in the text of the field that uses this type? Unfortunately, the GoDoc on the types don't get automatically included in the generated documentation of the fields that use it and I think this additional context could be useful for users.